aleph2_data_model/src/com/ikanow/aleph2/data_model/interfaces/data_access/IAccessContext.java

```diff
@@ -30,9 +30,43 @@
 /** A context library that is always passed to the AccessModules via injection 
  * @author acp
  */
-public interface IAccessContext {
 
-	//TODO (ALEPH-1):
+import com.google.inject.Injector;
+import com.ikanow.aleph2.data_model.interfaces.data_services.IColumnarService;
+import com.ikanow.aleph2.data_model.interfaces.data_services.IDocumentService;
+import com.ikanow.aleph2.data_model.interfaces.data_services.IGeospatialService;
+import com.ikanow.aleph2.data_model.interfaces.data_services.IGraphService;
+import com.ikanow.aleph2.data_model.interfaces.data_services.IManagementDbService;
+import com.ikanow.aleph2.data_model.interfaces.data_services.ISearchIndexService;
+import com.ikanow.aleph2.data_model.interfaces.data_services.IStorageService;
+import com.ikanow.aleph2.data_model.interfaces.data_services.ITemporalService;
+import com.ikanow.aleph2.data_model.interfaces.shared_services.ISecurityService;
+
+/**
+ * Gives an app access to all the currently configured data services.
+ * 
+ * @author Burch
+ *
+ */
+public interface IAccessContext {	
+	
+	//generic get service interface
+	public <I> I getDataService(@NonNull Class<I> service_clazz);
+	
+	//utility getters for common services
+	public IColumnarService getColumnarService();
+	public IDocumentService getDocumentService();
+	public IGeospatialService getGeospatialService();
+	public IGraphService getGraphService();	
+	public IManagementDbService getManagementDbService();
+	public ISearchIndexService getSearchIndexService();
+	public IStorageService getStorageIndexService();
+	public ITemporalService getTemporalService();
+	
+	//security service is related to data services
+	public ISecurityService getSecurityService();
+
+	public Injector getInjector();
```

Comment:
I need to add comments to all this, but the injector is used in that example I was sending around about using guice-servlet, you have to get the current injector and create a child from it with your apps injections so that you can inject the data services via @Inject tags